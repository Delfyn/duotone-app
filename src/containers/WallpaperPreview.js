/* @flow */

import React, { Component } from 'react';
import { createDuotoneImage } from 'react-duotone';
import Wallpaper from '../components/Wallpaper';

class WallpaperPreview extends Component {
  constructor() {
    super();
    this.state = {};
  }
  state = {
    preview: String,
  };
  componentWillReceiveProps(nextProps: any) {
    if (nextProps.originalFile) {
      this.setPreviewImage(nextProps);
    }
  }
  setPreviewImage(props: any) {
    const image = new Image();
    image.width = 15;
    image.height = 10;
    image.src = props.originalFile || this.props.originalFile;
    image.onload = () => {
      const primary = props.primary || this.props.primary;
      const secondary = props.secondary || this.props.secondary;
      this.setState({
        preview: createDuotoneImage(image, primary, secondary),
      });
    };
  }
  render() {
    const props = {
      ...this.props,
      ...this.state,
    };

    return <Wallpaper {...props} />;
  }
}

export default WallpaperPreview;
