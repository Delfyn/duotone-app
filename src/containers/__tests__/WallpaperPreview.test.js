import React from 'react';
import renderer from 'react-test-renderer';
import WallpaperPreview from '../WallpaperPreview';

test('WallpaperPreview renders correctly', () => {
  const component = renderer.create(
    <WallpaperPreview fileName="my-image.png" />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
