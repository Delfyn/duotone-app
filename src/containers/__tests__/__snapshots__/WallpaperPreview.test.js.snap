// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WallpaperPreview renders correctly 1`] = `
<div
  className="image-background-wrapper flex items-center"
>
  <div
    className="image-background z1"
    style={
      Object {
        "backgroundImage": null,
      }
    }
  />
</div>
`;
